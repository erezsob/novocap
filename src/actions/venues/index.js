// @flow
import * as types from './../../common/ActionTypes';
import { venuesService } from '../../services/services';
import { handleServerRequestError } from './../app';
import { LOCAL_STORAGE_VENUE_KEY } from './../../constants';

import type { Venues, Dispatch, SessionVenue } from './../../common/types';

export function fetchVenuesSuccess(venues: Venues) {
  return (dispatch: Dispatch) => {
    dispatch({ type: types.LOADING, loading: false });
    dispatch({
      type: types.FETCH_VENUES_SUCCESS,
      venues
    });
    const venue: SessionVenue = {
      id: venues[0].uuid,
      name: venues[0].name,
      currencyType: venues[0].currencyType
    };

    localStorage.setItem(LOCAL_STORAGE_VENUE_KEY, JSON.stringify(venue));
  };
}

export function fetchVenues() {
  return (dispatch: Dispatch) => {
    dispatch({ type: types.LOADING, loading: true });

    return venuesService()
      .then(response => response.json())
      .then(venues => dispatch(fetchVenuesSuccess(venues)))
      .catch(response => {
        dispatch(handleServerRequestError(response));
      });
  };
}
