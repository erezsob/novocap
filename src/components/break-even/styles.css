.container {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    height: 20px;
    width: 90%;
    padding-top: 20px;
    color: white;
    font-weight: bold;
    font-size: 20px;
}

.barContainer {
    margin: 20px;
    width: 150px;
    height: 150px;
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.title {
    display: flex;
    flex-direction: column;
    padding-top: 13px;
}

.date {
    font-size: 11px;
    font-weight: 300;
}

.addCosts {
    color: #fff;
    font-size: 19px;
}