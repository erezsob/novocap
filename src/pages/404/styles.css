@value blueDark, blueText from './../../common/variables.css';

.container {
    display: flex;
    height: 100vh;
    align-items: center;
    justify-content: center;
}

.textBox {
    text-align: center;
    color: blueText;
    width: 273px;
    min-height: 260px;
}

.bigText {
    font-size: 800%;
    color: blueDark;
}
