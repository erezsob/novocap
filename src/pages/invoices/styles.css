@value bpSMedium from './../../common/variables.css';

.controls {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    margin-top: 20px;
}

.controls > ul {
    padding: 20px 0 10px;
}

.controlsTop {
    composes: controls;
    justify-content: flex-end;
    margin: 0 0 10px;
}

.controlsTop > ul {
    padding: 10px 0;
}

@media(min-width: bpSMedium) and (min-device-width: bpSMedium) {
    .controls {
        height: 40px;
        flex-direction: row;
        padding: 20px 0;
        margin: 0;
    }

    .controls > * {
        padding: 0;
    }

    .controlsTop > ul {
        margin-bottom: 0;
    }
}
