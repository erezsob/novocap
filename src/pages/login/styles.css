@value washedOutWhite, red200, bpXSmall, bpMedium from './../../common/variables.css';

.container {
    display: flex;
    height: 100vh;
    align-items: center;
    justify-content: center;
    background: transparent url(../../img/login_bg.jpg) 20% center no-repeat;
    background-attachment: fixed;
    background-size: cover;
    position: relative;
}

@media (min-width: bpMedium) and (min-device-width: bpMedium) {
    .container {
        background-position-x: center;
    }
}

.loginBox {
    display: flex;
    align-items: center;
    flex-direction: column;
    background-color: washedOutWhite;
    width: 100%;
    height: 100%;
    justify-content: center;
    opacity: 0;
    transform: translateY(100px);
    transition: opacity 0.7s cubic-bezier(0.39, 0.575, 0.565, 1), transform 0.5s cubic-bezier(0.39, 0.575, 0.565, 1);
}

@media (min-width: bpXSmall) and (min-device-width: bpXSmall) {
    .loginBox {
        width: 273px;
        height: auto;
        min-height: 260px;
        justify-content: space-around;
        border-radius: 20px;
        padding: 26px 36px;
    }
}

.logo {
    max-width: 200px;
}

.boxEntered {
    opacity: 1;
}

.boxEntering {
    opacity: 0.01;
}

.boxEntered {
    transform: translateY(0);
}