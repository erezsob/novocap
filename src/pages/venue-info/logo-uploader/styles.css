@value defaultText, blueLight, offWhite, newBlack from '../../../common/variables.css';

.cloudLogo {
    color: lightGray;
}

.text1 {
    font-size: 16px;
    line-height: 20px;
    font-weight: bold;
    color: newBlack;
}

.text2,
.text3 {
    font-size: 11px;
    line-height: 14px;
}

.text2 {
    color: defaultText;
}

.text3 {
    color: blueLight;
}

.dashedBox {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-radius: 4px;
    border: 2px dashed offWhite;
    text-align: center;
    height: auto;
    padding: 10px 0;
}

@media (min-width: 960px) and (min-device-width: 960px) {
    .dashedBox {
        height: 245px;
        padding: 0;
    }
}