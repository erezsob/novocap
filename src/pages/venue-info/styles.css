.roundedWhiteBox {
    background-color: white;
    border-radius: 4px;
    padding: 12px 24px;
    width: 100%;
    box-sizing: border-box;
}

.group {
    margin-top: 10px;
}

.wrapper {
    display: flex;
    flex-wrap: wrap;
    width: 100%;
    box-sizing: border-box;
    margin: 0 -12px;
}

.extraPadding {
    padding: 12px 10px 0;
    width: 100%;
}

.map {
    border: 0;
    width: 100%;
    height: 245px;
}

.container {
    padding: 12px;
    display: flex;
    flex-wrap: wrap;
    flex: 0 0 auto;
    box-sizing: border-box;
}

.container50 {
    composes: container;
    flex-basis: 100%;
    max-width: 100%;
}

.container100 {
    composes: container;
    flex-basis: 100%;
    max-width: 100%;
}

@media (min-width: 960px) and (min-device-width: 960px) {
    .container50 {
        flex-basis: 50%;
        max-width: 50%;
    }

    .extraPadding {
        padding: 12px 0 0;
    }
}
