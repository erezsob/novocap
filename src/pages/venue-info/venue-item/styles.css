@value defaultText, newBlack, gray74 from '../../../common/variables.css';

.ellipsis {
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.box {
    composes: roundedWhiteBox from '../styles.css';
}

.label {
    composes: ellipsis;
    color: newBlack;
    font-size: 12px;
    font-weight: bold;
    margin-bottom: 10px;
}

.field {
    composes: ellipsis;
    font-size: 16px;
    height: 16px;
    color: defaultText;
    padding-bottom: 10px;
    border-bottom: 1px solid gray74;
}
