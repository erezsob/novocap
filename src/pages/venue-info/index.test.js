import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { VenuePage } from './';
import LogoUploader from './logo-uploader';
import { GOOGLE_MAPS_API_KEY } from './../../constants';
import venues from './../../../mockdata/venues.json';
import styles from './styles.css';

describe('<VenuePage />', () => {
  let wrapper;
  let spyComponentDidMount;
  let spyFetchVenues;

  beforeEach(() => {
    spyComponentDidMount = jest.spyOn(VenuePage.prototype, 'componentDidMount');
    spyFetchVenues = jest.fn();

    wrapper = shallow(
      <VenuePage venue={venues[0]} fetchVenues={spyFetchVenues} />
    );
  });

  afterEach(() => {
    spyComponentDidMount.mockClear();
    spyFetchVenues.mockClear();
  });

  it('renders properly', () => {
    const tree = renderer.create(
      <VenuePage venue={venues[0]} fetchVenues={spyFetchVenues} />
    );
    expect(tree.toJSON()).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });

  describe('Lifecycle methods', () => {
    test('componentDidMount()', () => {
      expect(spyComponentDidMount).toHaveBeenCalledTimes(1);
      expect(spyFetchVenues).toHaveBeenCalled();
    });
  });

  it('renders a <LogoUploader />', () => {
    expect(wrapper.find(LogoUploader)).toHaveLength(1);
  });

  it('renders a map with the right attributes', () => {
    const { address } = venues[0];
    expect(wrapper.find(`.${styles.map}`).props().src).toBe(
      `https://www.google.com/maps/embed/v1/place?key=${GOOGLE_MAPS_API_KEY}&q=${
        address.latitude
      },${address.longitude}`
    );
  });
});
