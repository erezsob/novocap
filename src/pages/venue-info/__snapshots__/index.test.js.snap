// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VenuePage /> renders properly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-24-24 MuiGrid-justify-xs-center-18"
    >
      <div
        className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38"
      >
        <div
          className="title"
        >
          General Information
        </div>
      </div>
      <div
        className="wrapper"
      >
        <div
          className="container50"
        >
          <div
            className="container100"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Venue Name
              </div>
              <div
                className="field"
              >
                CozyCorner
              </div>
            </div>
          </div>
          <div
            className="container100"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Created at
              </div>
              <div
                className="field"
              >
                Fri, Jan 1, 2016 12:00 AM
              </div>
            </div>
          </div>
          <div
            className="container100"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                ID
              </div>
              <div
                className="field"
              >
                b1169e68-da9a-11e7-a6c2-1b5d67d3d955
              </div>
            </div>
          </div>
        </div>
        <div
          className="container50"
        >
          <div
            className="extraPadding"
          >
            <div
              className="roundedWhiteBox"
            >
              <div
                className="dashedBox"
              >
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-91 cloudLogo"
                  color={undefined}
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                  />
                </svg>
                <span
                  className="text1"
                >
                  Drag & Drop your logo here!
                </span>
                <span
                  className="text2"
                >
                  or
                </span>
                <span
                  className="text3"
                >
                  browse your computer
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-24-24 MuiGrid-justify-xs-center-18"
    >
      <div
        className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38"
      >
        <div
          className="title"
        >
          Address
        </div>
      </div>
      <div
        className="wrapper"
      >
        <div
          className="container50"
        >
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Address (Line 1)
              </div>
              <div
                className="field"
              >
                Evergreen
              </div>
            </div>
          </div>
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Address (Line 2)
              </div>
              <div
                className="field"
              >
                
              </div>
            </div>
          </div>
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Zip Code
              </div>
              <div
                className="field"
              >
                33445
              </div>
            </div>
          </div>
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                City
              </div>
              <div
                className="field"
              >
                Las Vegas
              </div>
            </div>
          </div>
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                State
              </div>
              <div
                className="field"
              >
                NV
              </div>
            </div>
          </div>
          <div
            className="container50"
          >
            <div
              className="box"
            >
              <div
                className="label"
              >
                Country
              </div>
              <div
                className="field"
              >
                USA
              </div>
            </div>
          </div>
        </div>
        <div
          className="container50"
        >
          <div
            className="extraPadding"
          >
            <div
              className="roundedWhiteBox"
            >
              <iframe
                className="map"
                frameBorder="0"
                src="https://www.google.com/maps/embed/v1/place?key=AIzaSyDQeH2cmhTOd435kRLbKO0iqpSOssScAi0&q=0,0"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="container"
  >
    <div
      className="MuiGrid-typeContainer-1 MuiGrid-spacing-xs-24-24 MuiGrid-justify-xs-center-18"
    >
      <div
        className="MuiGrid-typeItem-2 MuiGrid-grid-xs-12-38"
      >
        <div
          className="title"
        >
          Currency
        </div>
      </div>
      <div
        className="wrapper"
      >
        <div
          className="container50"
        >
          <div
            className="box"
          >
            <div
              className="label"
            >
              Country
            </div>
            <div
              className="field"
            >
              EUR
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
